src/plugins/condition/criterion/TelemetryCriterion.js

```diff
@@ -0,0 +1,83 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2019, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+import * as EventEmitter from 'eventemitter3';
+
+export default class TelemetryCriterion extends EventEmitter {
+
+    constructor(telemetryDomainObject, openmct) {
+        super();
+
+        this.telemetryObject = telemetryDomainObject;
+        this.openmct = openmct;
+        this.telemetryAPI = this.openmct.telemetry;
+        this.subscription = null;
+        this.telemetryObjectIdAsString = null;
+        this.telemetryMetadata = null;
+        if (this.telemetryAPI.isTelemetryObject(this.telemetryObject)) {
+            this.telemetryObjectIdAsString = this.openmct.objects.makeKeyString(this.telemetryObject.identifier);
+        }
+    }
+
+    handleSubscription(datum) {
+        //data is telemetry values, error
+        //how do I get data here?
+        this.emitResult(this.normalizeData(datum));
+    }
+
+    //TODO: Revisit this logic
+    normalizeData(datum) {
+        return {
+            [datum.key]: datum[datum.source]
+        }
+    }
+
+    emitResult(data, error) {
+        this.emit('criterion::Update', {
```

Comment:
I don't really follow this. Would appreciate some help understanding it.

Reply:
It's emitting an event saying that this criterion has been updated. Any listeners will handle the data provided by the event.

Reply:
How about emitting 'criterionUpdated' event? Emitting update indicates that listeners should update something.

Reply:
Ok I've changed this - makes sense.